### PR TITLE
TVA-1072: Implements financial provider constants api adapter

### PR DESCRIPTION
**Pull request by kostasakrivos**
### Exclude spring boot default logging test
We want to exclude spring-boot's default implementation of logging as we
want to a simpler for tests. This change makes sure that this happens
only in test configurations rather than in the whole set of modules.

### Moved driving adapter in another package
As this module will include both adapters of the component, we moved the
driving adapter into its own package so that it gives space for the
financial provider constants api adapter.

### Implemented http driven adapter
Implemented the http adapter of financial provider constants driven port
. Feign, by default, throws exceptions for all http status codes, but we
opted to recover from a `NotFound extends FeignClientException` to
correctly return the expected result from a client error.

For testing, we built a mock decorator, that delegates the call to the
production http adapter, but reaches a mocked http server.

### CODE REVIEW

**Comment by ytree-admin** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **kostasakrivos**
You can assign the PR to them by writing `/assign @kostasakrivos` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=y-tree-limited%2Fms-dob-fwk).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/y-tree-limited/ms-dob-fwk/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: dob-status-salesforce-event-consumer/src/main/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/FinancialProviderConstantsHttpApi.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.ytree.dob_fwk.status.crm_event_consumer.application.port.driven.model.ProviderConstantId
+import com.ytree.dob_fwk.status.crm_event_consumer.domain.ProviderName
+import com.ytree.lib.constants.api.provider.ConstantProviderApi
+import feign.FeignException.NotFound
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class FinancialProviderConstantsHttpApi(
+    @Autowired private val constantProviderApi: ConstantProviderApi
```
Comment (Petlyuk-Vladimir) [inline dob-status-salesforce-event-consumer/src/main/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/FinancialProviderConstantsHttpApi.kt:14]: you can remove @Autowired here

File: dob-status-salesforce-event-consumer/src/main/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/FinancialProviderConstantsHttpApiConfig.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import feign.Feign
+import feign.jackson.JacksonDecoder
+import feign.jackson.JacksonEncoder
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+open class FinancialProviderConstantsHttpApiConfig(
+    @Value("\${msConstants.baseUrl}")
```
Comment (Petlyuk-Vladimir) [inline dob-status-salesforce-event-consumer/src/main/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/FinancialProviderConstantsHttpApiConfig.kt:14]: check plz naming for base url in montrose-dev2 properties 
https://gyazo.com/93baa4e718c5221a8f3518532c0d08b5

File: dob-status-salesforce-event-consumer/src/test/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/ConstantProviderApiWireMockDecorator.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import com.ytree.dob_fwk.status.crm_event_consumer.application.port.driven.model.FinancialProviderConstant
+import com.ytree.dob_fwk.status.crm_event_consumer.domain.ProviderName
+import com.ytree.dob_fwk.status.crm_event_consumer.test_fixtures.TestingFinancialProviderConstantsApi
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import java.net.URI
+
+@Component
+class ConstantProviderApiWireMockDecorator(
+    @Autowired private val baseUrl: URI,
```
Comment (Petlyuk-Vladimir) [inline dob-status-salesforce-event-consumer/src/test/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/ConstantProviderApiWireMockDecorator.kt:17]: same you can remove @Autowired

File: dob-status-salesforce-event-consumer/src/test/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/HandleCrmEventViaFinancialProviderConstantsHttpApi.kt [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+import org.springframework.context.annotation.Import
+import org.springframework.test.context.ActiveProfiles
+import java.net.URI
+
+@SpringBootTest(classes = [TestApp::class], properties = ["msConstants.baseUrl=$BASE_URL_RAW"])
+@ActiveProfiles("test")
+@Import(TestConfig::class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class HandleCrmEventViaFinancialProviderConstantsHttpApi(
+    @Autowired
```
[the remark this answers is not in the document]
Comment (kostasakrivos) [reply dob-status-salesforce-event-consumer/src/test/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/HandleCrmEventViaFinancialProviderConstantsHttpApi.kt:36]: I would like to keep `@Autowired` to indicate that the dependency would be brought in by spring framework. For me it helps for readability purposes. If you don't mind, for me currently it is useful.

File: dob-status-salesforce-event-consumer/src/main/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/FinancialProviderConstantsHttpApi.kt (hunk 1 above)
Comment (kostasakrivos) [reply dob-status-salesforce-event-consumer/src/main/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/FinancialProviderConstantsHttpApi.kt:14]: I would like to keep `@Autowired` to indicate that the dependency would be brought in by spring framework. For me it helps for readability purposes. If you don't mind, for me currently it is useful.

File: dob-status-salesforce-event-consumer/src/test/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/ConstantProviderApiWireMockDecorator.kt (hunk 3 above)
Comment (kostasakrivos) [reply dob-status-salesforce-event-consumer/src/test/kotlin/com/ytree/dob_fwk/status/sf_event_consumer/adapter/driven/ConstantProviderApiWireMockDecorator.kt:17]: I would like to keep `@Autowired` to indicate that the dependency would be brought in by spring framework. For me it helps for readability purposes. If you don't mind, for me currently it is useful.

**Review by sn3jkeee — DISMISSED** [review]

**Comment by ytree-admin** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **kostasakrivos**
You can assign the PR to them by writing `/assign @kostasakrivos` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=y-tree-limited%2Fms-dob-fwk).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/y-tree-limited/ms-dob-fwk/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ytree-admin** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **kostasakrivos**
You can assign the PR to them by writing `/assign @kostasakrivos` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=y-tree-limited%2Fms-dob-fwk).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/y-tree-limited/ms-dob-fwk/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by Petlyuk-Vladimir — APPROVED** [review]

**Comment by ytree-admin** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[Petlyuk-Vladimir](https://github.com/y-tree-limited/ms-dob-fwk/pull/10#pullrequestreview-1379164154 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=y-tree-limited%2Fms-dob-fwk).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/y-tree-limited/ms-dob-fwk/blob/main/OWNERS)~~ [Petlyuk-Vladimir]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**